HugeCTR/include/hps/hash_map_backend_detail.hpp

```diff
@@ -165,9 +172,16 @@ namespace HugeCTR {
       payload.value = part.value_slots.back();                                               \
       part.value_slots.pop_back();                                                           \
       ++num_inserts;                                                                         \
+      lock.unlock();                                                                         \
     }                                                                                        \
                                                                                              \
+    if (payload.lck.load(std::memory_order_relaxed) != -1) {                                 \
```

Comment:
This looks like it could work. It is a little bit fuzzy. Can you clarify the intended lifecycle of `lck`. What value should follow after what?